File: javaSample/src/main/java/com/cleveradssolutions/sampleapp/nativead/NativeAdTemplateActivity.java
package com.cleveradssolutions.sampleapp.nativead;

import static com.cleveradssolutions.sampleapp.SampleApplication.CAS_ID;

import android.graphics.Color;
import android.os.Bundle;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;

import com.cleveradssolutions.sampleapp.R;
import com.cleveradssolutions.sampleapp.support.BackNavigationActivity;
import com.cleveradssolutions.sampleapp.support.Util;
import com.cleveradssolutions.sdk.AdContentInfo;
import com.cleveradssolutions.sdk.nativead.AdChoicesPlacement;
import com.cleveradssolutions.sdk.nativead.CASNativeLoader;
import com.cleveradssolutions.sdk.nativead.CASNativeView;
import com.cleveradssolutions.sdk.nativead.NativeAdContent;
import com.cleveradssolutions.sdk.nativead.NativeAdContentCallback;
import com.cleversolutions.ads.AdError;
import com.cleversolutions.ads.AdSize;

public class NativeAdTemplateActivity extends BackNavigationActivity {

    private CASNativeView adView;

    private final NativeAdContentCallback nativeAdCallback = new NativeAdContentCallback() {

        @Override
        public void onNativeAdLoaded(@NonNull NativeAdContent nativeAd, @NonNull AdContentInfo ad) {
            Util.toast(NativeAdTemplateActivity.this, "Native Ad loaded");
            FrameLayout container = findViewById(R.id.container);
            inflateNativeAdView(container);
            registerNativeAdContent(nativeAd);
        }

        @Override
        public void onNativeAdFailedToLoad(@NonNull AdError error) {
            Util.toastError(NativeAdTemplateActivity.this, "Native Ad failed to load: " + error.getMessage());
        }

        @Override
        public void onNativeAdFailedToShow(@NonNull NativeAdContent nativeAd, @NonNull AdError error) {
            Util.toastError(NativeAdTemplateActivity.this, "Native Ad show failed: " + error.getMessage());
        }

        @Override
        public void onNativeAdClicked(@NonNull NativeAdContent nativeAd, @NonNull AdContentInfo ad) {
            Util.toast(NativeAdTemplateActivity.this, "Native Ad clicked");
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.container_activity);
        setTitle(R.string.adFormats_nativeAdTemplate);

        loadNativeAd();
    }

    private void loadNativeAd() {
        CASNativeLoader adLoader = new CASNativeLoader(this, CAS_ID, nativeAdCallback);
        adLoader.setAdChoicesPlacement(AdChoicesPlacement.TOP_LEFT);
        adLoader.setStartVideoMuted(true);

        adLoader.load();
    }

    private void registerNativeAdContent(NativeAdContent nativeAd) {
        if (adView != null) {
            adView.bindAdContent(nativeAd);
        }
    }

    private void inflateNativeAdView(ViewGroup container) {
        CASNativeView adView = new CASNativeView(this);
        this.adView = adView;
        AdSize size = AdSize.MEDIUM_RECTANGLE;
        adView.setAdTemplateSize(size);

        customizeAdViewAppearance(adView);

        container.addView(adView);
    }

    private void customizeAdViewAppearance(CASNativeView adView) {
        // Default values are shown below:
//        adView.setBackgroundColor(Color.WHITE);

//        if (adView.getCallToActionView() != null) {
//            adView.getCallToActionView().setBackgroundColor(Color.BLUE);
//        }

//        if (adView.getHeadlineView() != null) {
//            adView.getHeadlineView().setTextColor(Color.parseColor("#80000000"));
//        }
    }

}

File: kotlinSample/src/main/java/com/cleveradssolutions/sampleapp/native/NativeAdTemplateActivity.kt
package com.cleveradssolutions.sampleapp.native

import android.graphics.Color
import android.os.Bundle
import android.view.ViewGroup
import android.widget.FrameLayout
import com.cleveradssolutions.sampleapp.R
import com.cleveradssolutions.sampleapp.SampleApplication.Companion.CAS_ID
import com.cleveradssolutions.sampleapp.util.BackNavigationActivity
import com.cleveradssolutions.sampleapp.util.toast
import com.cleveradssolutions.sampleapp.util.toastError
import com.cleveradssolutions.sdk.AdContentInfo
import com.cleveradssolutions.sdk.nativead.AdChoicesPlacement
import com.cleveradssolutions.sdk.nativead.CASNativeLoader
import com.cleveradssolutions.sdk.nativead.CASNativeView
import com.cleveradssolutions.sdk.nativead.NativeAdContent
import com.cleveradssolutions.sdk.nativead.NativeAdContentCallback
import com.cleversolutions.ads.AdError
import com.cleversolutions.ads.AdSize

class NativeAdTemplateActivity : BackNavigationActivity() {

    private var adView: CASNativeView? = null

    private val nativeAdCallback: NativeAdContentCallback = object : NativeAdContentCallback() {

        override fun onNativeAdLoaded(nativeAd: NativeAdContent, ad: AdContentInfo) {
            toast("Native Ad loaded")
            val container = findViewById<FrameLayout>(R.id.container)
            inflateNativeAdView(container)
            registerNativeAdContent(nativeAd)
        }

        override fun onNativeAdFailedToLoad(error: AdError) {
            toastError("Native Ad failed to load: ${error.message}")
        }

        override fun onNativeAdFailedToShow(nativeAd: NativeAdContent, error: AdError) {
            toastError("Native Ad show failed: ${error.message}")
        }

        override fun onNativeAdClicked(nativeAd: NativeAdContent, ad: AdContentInfo) {
            toast("Native Ad clicked")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.container_activity)
        setTitle(R.string.adFormats_nativeAdTemplate)

        loadNativeAd()
    }

    private fun loadNativeAd() {
        val adLoader = CASNativeLoader(this, CAS_ID, nativeAdCallback)
        adLoader.adChoicesPlacement = AdChoicesPlacement.TOP_LEFT
        adLoader.isStartVideoMuted = true

        adLoader.load()
    }

    private fun registerNativeAdContent(nativeAd: NativeAdContent) {
        adView?.setNativeAd(nativeAd)
    }

    private fun inflateNativeAdView(container: ViewGroup) {
        val adView = CASNativeView(this)
        this.adView = adView
        val size = AdSize.MEDIUM_RECTANGLE
        adView.setAdTemplateSize(size)

        customizeAdViewAppearance(adView)

        container.addView(adView)
    }

    private fun customizeAdViewAppearance(adView: CASNativeView) {
        // Default values are shown below:
        adView.setBackgroundColor(Color.WHITE)

        adView.callToActionView?.setBackgroundResource(com.cleveradssolutions.sdk.android.R.drawable.cas_rounded_button)
        adView.headlineView?.setTextColor(Color.parseColor("#80000000"))
    }

}